test/routers/user.test.js

```diff
@@ -0,0 +1,58 @@
+const request = require('supertest');
+const express = require('express');
+const router = require('../../routes/users');
+const ApiError = require('../../utils/api-error');
+const { users } = require('../../models');
+require('dotenv').config();
+
+const app = express();
+app.use(express.json());
+app.use('/users', router);
+
+jest.mock('../../models');
+jest.mock('jsonwebtoken', () => ({
+  sign: jest.fn(() => 'mockedToken'),  
+}));
+
+describe('POST /users/register', () => {
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should create a new user and return a success message and token', async () => {
+    users.create.mockResolvedValue({ id: 1, name: 'TestUser', email: 'test@example.com' });
+    
```

Comment:
you only mocked users.create call. you should mock:
    const existingName = await users.findOne({ where: { name } });
    const newUser = await users.create({